src/health-monitor.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+const log = require('debugnyan')('process-manager:health-monitor');
+
+/**
+ * `HealthMonitor`.
+ */
+
+class HealthMonitor {
+  /**
+   * Constructor.
+   */
+
+  constructor() {
+    this.checks = [];
+    this.globalState = HealthMonitor.states.UNKNOWN;
+    this.states = {};
+  }
+
+  /**
+   * Add health check.
+   */
+
+  addCheck({ handler, id, interval = 5000 }) {
+    if (this.states[id]) {
+      throw new Error('Cannot add handler since it would overwrite an existing one');
+    }
+
+    this.states[id] = HealthMonitor.states.UNKNOWN;
+
+    const check = setInterval(async () => {
```

Comment:
I’m not fond of setInterval with async functions, can we change this?

Reply:
If I understood correctly, your misgivings have been fixed.